bin/workflows/pycbc_make_faithsim_workflow

```diff
@@ -0,0 +1,187 @@
+
+        node.new_output_file_opt(
+            analysis_time, ".xml", "--output", tags=self.tags + extra_tags
+        )
+        return node
+
+
+class FaithsimExecutable(wf.Executable):
+    """Class for running pycbc_faithsim"""
+
+    current_retention_level = wf.Executable.ALL_TRIGGERS
+
+    def create_node(self, analysis_time, param_file, extra_tags=None):
+        if extra_tags is None:
+            extra_tags = []
+        node = wf.Executable.create_node(self)
+
+        node.add_input_opt("--param-file", param_file)
+
+        node.new_output_file_opt(
+            analysis_time, ".dat", "--match-file", tags=self.tags + extra_tags
+        )
+        return node
+
+
+class CollectResultsExecutable(wf.Executable):
+    """Class for collecting the results of the faithsim script"""
+
+    current_retention_level = wf.Executable.ALL_TRIGGERS
+
+    def create_node(self, analysis_time, match_list, bank_list, extra_tags=None):
+        if extra_tags is None:
+            extra_tags = []
+        node = wf.Executable.create_node(self)
+
+        for match in match_list:
+            node.add_input_opt("--inputs-match", match)
+        for bank in bank_list:
+            node.add_input_opt("--inputs-bank", bank)
+
+        node.new_output_file_opt(
+            analysis_time, ".dat", "--output", tags=self.tags + extra_tags
+        )
+
+        return node
+
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument("--version", action="version", version=__version__)
+wf.add_workflow_command_line_group(parser)
+wf.add_workflow_settings_cli(parser)
+args = parser.parse_args()
+
+workflow = wf.Workflow(args)
+
+config_path = os.path.abspath(args.config_files[0])
+
+wf.makedir(args.output_dir)
+os.chdir(args.output_dir)
+
+wf.makedir("match")
+wf.makedir("collect_results")
+wf.makedir("injections")
+wf.makedir("bank")
+wf.makedir("plots")
+
+num_banks = workflow.cp.get("splitbank", "num_banks")
+
+injections_exe = CreateInjectionExecutable(
+    workflow.cp,
+    "lalapps_inspinj",
+    ifos=["H1", "L1", "V1"],
+    out_dir="injections",
+    tags=["inj"],
+)
+
+inj_node = injections_exe.create_node(workflow.analysis_time)
+workflow += inj_node
+
+inj = inj_node.output_files[0]
+
+split_exe = wf.PycbcSplitBankXmlExecutable(
+    workflow.cp, "pycbc_splitbank", num_banks=num_banks, out_dir="bank"
+)
+spltbank_node = split_exe.create_node(inj)
+
+workflow += spltbank_node
+
+faithsim_exe = FaithsimExecutable(
+    workflow.cp, "pycbc_faithsim", ifos=["H1", "L1", "V1"], out_dir="match"
+)
+
+collect_exe = CollectResultsExecutable(
+    workflow.cp,
+    "pycbc_faithsim_collect_results",
+    ifos=["H1", "L1", "V1"],
+    out_dir="collect_results",
+)
+
+faithsim_files = wf.FileList([])
+
+for i in range(int(num_banks)):
+    bank = spltbank_node.output_files[i]
+    faithsim_node = faithsim_exe.create_node(
+        workflow.analysis_time,
+        param_file=bank,
+        extra_tags=["match-{i}"],
```

Comment:
You must not use dashses `-` in tag names! Dashes are used as a separator so this can confuse things. Instead use "match_{i}" or "match{i}"